packages/blade/src/components/FileUpload/FileUpload.web.tsx

```diff
@@ -0,0 +1,323 @@
+import { Text } from '~components/Typography';
+import type { BladeElementRef } from '~utils/types';
+import { getHintType } from '~components/Input/BaseInput/BaseInput';
+import { makeAccessible } from '~utils/makeAccessible';
+
+const MemoizedFileUploadItem = memo(FileUploadItem);
+
+const _FileUpload: React.ForwardRefRenderFunction<BladeElementRef, FileUploadProps> = (
+  {
+    name,
+    accept,
+    uploadType = 'single',
+    onChange,
+    onPreview,
+    onRemove,
+    onCancel,
+    onDrop,
+    isDisabled,
+    isRequired,
+    necessityIndicator,
+    fileList,
+    testID,
+    label,
+    labelPosition = 'top',
+    accessibilityLabel,
+    validationState,
+    helpText,
+    errorText,
+    maxCount,
+    maxSize,
+    ...styledProps
+  },
+  ref,
+): React.ReactElement => {
+  const { platform } = useTheme();
+  const [selectedFiles, setSelectedFiles] = useState<BladeFileList>(fileList ?? []);
+  const [errorMessage, setErrorMessage] = useState(errorText);
+  const [internalValidationState, setInternalValidationState] = useState('none');
+  const [isActive, setIsActive] = useState(false);
+
+  const isMultiple = uploadType === 'multiple';
+  const inputLabelPosition = platform === 'onMobile' ? 'top' : labelPosition;
+  const isLabelLeftPositioned = inputLabelPosition === 'left';
+  const willRenderHintText =
+    !isMultiple && selectedFiles.length === 1 ? false : Boolean(helpText) || Boolean(errorMessage);
+
+  const showError = validationState === 'error' || internalValidationState === 'error';
+  const showHelpText = !showError && helpText;
+  const accessibilityText =
+    accessibilityLabel ?? `,${showError ? errorText : ''} ${showHelpText ? helpText : ''}`;
+  const { inputId, labelId, helpTextId, errorTextId } = useFormId('fileinput');
+
+  const accessibilityProps = makeAccessible({
+    required: Boolean(isRequired),
+    invalid: Boolean(showError),
+    disabled: Boolean(isDisabled),
+    describedBy: helpTextId,
+  });
+
+  // In control mode attach a unique id to each file if not provided
+  useMemo(() => {
+    for (const file of selectedFiles) {
+      if (!file.id) {
+        file.id = `${new Date().getTime().toString()}${Math.floor(Math.random() * 1000000)}`;
+      }
+    }
+  }, [selectedFiles]);
+
+  const handleFilesChange = useCallback((inputFiles: BladeFileList) => {
+    setSelectedFiles((prevFiles) => {
+      if (prevFiles.length > 0) {
+        const allFiles = [...prevFiles, ...inputFiles];
+        return allFiles;
+      }
+
+      return inputFiles;
+    });
+  }, []);
+
+  const validateFiles = (inputFiles: BladeFileList, allFiles: BladeFileList): boolean => {
+    if (uploadType === 'single' && inputFiles.length > 1) {
+      setErrorMessage('You can upload only one file.');
+      setInternalValidationState('error');
+      return true;
+    }
+
+    if (maxCount && allFiles.length > maxCount) {
+      setErrorMessage(`You can't upload more than ${maxCount} files.`);
```

Comment:
These messages are hardcoded? 

What if product wants to change these?

Reply:
These are the default validations messages, giving users control to customize this might complex the logic. We can improve if any use-case comes up. Additional validations & messages can be set via onChange & errorText props.

Reply:
When users pass errorText, will this error message be overriden? or both error messages exist at different places?

Reply:
`errorText` will override these.